source/synthDrivers/sapi5.py

```diff
@@ -88,13 +106,13 @@ class SapiSink(object):
 	"""
 
 	def __init__(self, synth):
-		self.synth = synth
+		self.synthRef = weakref.ref(synth)
 
 	def Bookmark(self, streamNum, pos, bookmark, bookmarkId):
-		synthIndexReached.notify(synth=self.synth, index=bookmarkId)
+		synthIndexReached.notify(synth=self.synthRef(), index=bookmarkId)

```

Comment:
Here (and in `EndStream`), for completeness I would like to see this check for `None` returned by `synthRef()`. This `SapiSink` shouldn't exist if the synth doesn't, but it's simple defensive programming in case of design changes.

Reply:
Done as requested.